docs/azure/integrations-overview.md

```diff
@@ -1,15 +1,39 @@
 
-## Add connection to existing Azure resources
+## Naming conventions and APIs for expressing Azure resources
+
+The distributed application builder, part of the [app host](../fundamentals/app-host-overview.md), employs the builder pattern, allowing consumers to `AddAzure*` resources to the [_app model_](../fundamentals/app-host-overview.md#terminology). In addition to simply adding resources, developers can also express configuration and how said resource behave in various execution contexts. Azure hosting integrations provide APIs to specify how these resources should be "published" and "run".
+
+### App host run modes
+
+When the app host executes, the [_execution context_](../fundamentals/app-host-overview.md#execution-context) determines whether the app host is in <xref:Aspire.Hosting.DistributedApplicationOperation.Run> or <xref:Aspire.Hosting.DistributedApplicationOperation.Publish> mode. The naming conventions for these APIs indicate the intended action for the resource. For more information on execution modes, see [Execution context](../fundamentals/app-host-overview.md#execution-context).
+
+The following table summarizes the naming conventions used to express Azure resources:
+
+| API | Description |
+|--|--|
+| `PublishAsConnectionString` | Changes the resource to be published as a connection string reference in the manifest. |
+| `PublishAsContainer` | Changes the resource to be published as a container in the manifest. |
+| `PublishAsExisting` | Marks the resource as an existing resource when the application is deployed. |
+| `RunAsContainer` | Configures the Azure resource to run locally in a container. For more information, see [Local containers](#local-containers). |
+| `RunAsEmulator` | Configures the Azure resource to be emulated. For more information, see [Local emulators](#local-emulators). |
+| `RunAsExisting` | Marks the resource as an existing resource when the application is running. |
+
+> [!NOTE]
+> Not all APIs are available on all Azure resources. For example, some Azure resources can be containerized or emulated, while others cannot.
+
+When it comes to specifying whether an Azure resource in the app model is an existing resource, you can use the `AsExisting` API. The `AsExisting` method marks the resource as an existing resource in both run and publish modes. You can query whether a resource has been marked as an existing resource, by calling the `IsExisting` extension method on the <xref:Aspire.Hosting.ApplicationModel.IResource>. For more information, see [Mark Azure resources as existing](#mark-azure-resources-as-existing).
```

Comment:
I feel like `AsExisting` should be in the table above. Was there a reason why it isn't?

Reply:
Table would be fine. We'd also want somewhere to capture the info I shared in https://github.com/dotnet/docs-aspire/pull/2642#discussion_r1964239088. A table might get a little too stuffy if we added all that?

Reply:
I think the info in https://github.com/dotnet/docs-aspire/pull/2642#discussion_r1964239088 belongs in a paragraph form here. 

Having `AsExisting` in the table calls attention to it and puts it on the same level as RunAsExisting and PublishAsExisting. I think it belongs in the same spot(s).